### PR TITLE
Update to Go 1.18

### PR DESCRIPTION
**Pull request by javierhonduco**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/parca-dev/parca-agent
 
-go 1.17
+go 1.18
```
Comment (kakkoyun) [inline go.mod:3]: This is actually not needed. This specified minimum required version to compile this module/codebase, if we don't need a specific version from 1.18 we shouldn't update this.

**Review by kakkoyun — CHANGES_REQUESTED** [review]
LGTM. Thanks for taking care of it.

I only pointed one issue.

File: go.mod (hunk 1 above)
Comment (javierhonduco) [reply go.mod:3]: I think we need it for https://github.com/parca-dev/parca-agent/pull/341 (https://tip.golang.org/doc/go1.18 says that `Settings` was made public in that release)

Comment (kakkoyun) [reply go.mod:3]: Ah, then yes. Sorry I've missed that. It'd have been better to add these changes to that PR then.

**Review by kakkoyun — APPROVED** [review]

**Pull request merged**